### PR TITLE
Admin data tables vue

### PR DESCRIPTION
**Pull request by dannon**
Cherrypick of relevant commit applied to dev cleanly;  #5191, minor cleanup.

### CODE REVIEW

File: client/galaxy/scripts/apps/admin.js [hunk 1]
```diff
@@ -96,9 +97,15 @@ window.app = function app(options, bootstrapped) {
         },
 
         show_user_api_keys: function() {
-            var vuemount = document.createElement("div");
-            this.page.display(vuemount);
-            new Vue(UserAPIKeys).$mount(vuemount);
+            var vueMount = document.createElement("div");
+            this.page.display(vueMount);
+            new Vue(UserAPIKeys).$mount(vueMount);
```
Comment (guerler) [inline client/galaxy/scripts/apps/admin.js:96]: Can we augment the `display` caller such that it does the mounting for Vue-components internally? The call might look like this: `this.page.display(new Vue(UserAPIKeys))`.

Comment (dannon) [reply client/galaxy/scripts/apps/admin.js:96]: @guerler Yeah, that's on my list (I mentioned this in a previous Vue PR).  If you want to see that prior to merge of this PR, I guess I can go ahead and add something simple for full-page vue components.

Comment (guerler) [reply client/galaxy/scripts/apps/admin.js:96]: Cool. Its up to you.

**Comment by bgruening** [comment]
Cool! @dannon @guerler is this the right way to go for grids? Or do you plan to use an official vue-grid component? Is anyone working on this?

**Comment by guerler** [comment]
Admin tables are very minimal i.e. there are no operations, options and redirects to forms. Whichever solution we find for regular grids should also be applicable to these type of table views.

**Comment by dannon** [comment]
@bgruening Yeah, this was a step forward for these grids in particular, but the big shift will happen when we swap the `mvc/grid-*` itself to a vue-grid component.  Then we can update these as well.

**Pull request merged**